app/controllers/active_storage/blobs_controller.rb

```diff
@@ -29,6 +36,17 @@ class ActiveStorage::BlobsController < ActiveStorage::BaseController
   def check_user
     if current_user.nil?
       redirect_to new_user_session_path
+    else
+      # current_organisation is not available here
+      org = Organisation.find(
+        ActiveStorage::Attachment
+          .find_by(blob: @blob)
+          .record_id
+      )
```

Comment:
How performant is this, do you know?
Personally I am not _too_ concerned if it's a bit slow, if we think we're going to replace this anyway...

Reply:
It is an extra query, you're right, but from testing it it has very little impact on user experience. ActiveStorage already makes a request to the database in the first place, and since this is fetching a single record based on an SQL-indexed-column I don't think it would incur too much delay.

Reply:
cool, just making sure it is searching for id or hash over an index